packages/service-library/src/servicelib/long_running_tasks/task.py

```diff
@@ -278,25 +310,43 @@ async def _status_update(self) -> None:
                     # already done and updatet data in redis
                     continue
 
-                # update and store in Redis
-                task_data.is_done = is_done
-
+                result_field: ResultField | None = None
                 # get task result
                 try:
-                    task_data.result_field = ResultField(
-                        result=object_to_string(task.result())
-                    )
+                    result_field = ResultField(str_result=dumps(task.result()))
                 except asyncio.InvalidStateError:
                     # task was not completed try again next time and see if it is done
                     continue
                 except asyncio.CancelledError:
-                    task_data.result_field = ResultField(
-                        error=object_to_string(TaskCancelledError(task_id=task_id))
+                    result_field = ResultField(
+                        str_error=dumps(TaskCancelledError(task_id=task_id))
                     )
                 except Exception as e:  # pylint:disable=broad-except
-                    task_data.result_field = ResultField(error=object_to_string(e))
+                    allowed_errors = TaskRegistry.get_allowed_errors(
+                        task_data.registered_task_name
+                    )
+                    if type(e) not in allowed_errors:
+                        _logger.exception(
+                            **create_troubleshootting_log_kwargs(
+                                (
+                                    f"Execution of {task_id=} finished with unexpected error, "
+                                    f"only the following are {allowed_errors=} are permitted"
+                                ),
+                                error=e,
+                                error_context={
```

Comment:
TIP: if this class would be a dataclass, you cnould add `self` in the error_context and get an automatic representation of all its members

Reply:
That's a nice feature, but here it would not be that useful, the context is complete as is.